python-package/lightgbm/basic.py

```diff
@@ -2507,13 +2507,13 @@ def _compare_params_for_warning(
         compare_result : bool
           Returns whether two dictionaries with params are equal.
         """
-        for k in other_params:
+        for k, v in other_params.items():
             if k not in ignore_keys:
-                if k not in params or params[k] != other_params[k]:
+                if k not in params or params[k] != v:
                     return False
-        for k in params:
+        for k, v in params.items():
```

Comment:
Fix [PLC0206](https://docs.astral.sh/ruff/rules/dict-index-missing-items/#dict-index-missing-items-plc0206) in new Ruff version.

Reply:
oh wow, I'm impressed that a linter could look in this far and find something like this! Really nice.